src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs

```diff
@@ -485,6 +486,33 @@ private void SetOptionValue<T>(string name, T value)
             {
                 Options.Add(name, value);
             }
-        }
+        }
+
+        public bool IsComparableTo(ConnectionDetails other)

```

Comment:
I originally did. However, it's incorrect to say it's the same and this shouldn't be used in dictionary / list handling of the objects. We ignore a number of properties and fudge the treatment of `DatabaseName` due to the fact an empty name could mean any DB due to SQL's default user DB preferences.